pkg/ruler/manager.go

```diff
@@ -142,8 +146,12 @@ func (r *DefaultMultiTenantManager) SyncRuleGroups(ctx context.Context, ruleGrou
 	r.managersTotal.Set(float64(len(r.userManagers)))
 }
 
+func (r *DefaultMultiTenantManager) BackUpRuleGroups(ctx context.Context, ruleGroups map[string]rulespb.RuleGroupList) {
```

Comment:
I might missed it , but where do we call this function?

Reply:
We call it inside syncRules https://github.com/cortexproject/cortex/blob/f577e38b8347e0993c4928679da17761f6686710/pkg/ruler/ruler.go#L594